jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ResourceSelector.kt

```diff
@@ -19,14 +20,28 @@ import software.aws.toolkits.jetbrains.utils.CompatibilityUtils
 import software.aws.toolkits.jetbrains.utils.notifyError
 import software.aws.toolkits.jetbrains.utils.ui.find
 import software.aws.toolkits.resources.message
-import javax.swing.DefaultComboBoxModel
 
-typealias Selector<T> = Either<Any?, (T) -> Boolean>
+private typealias Selector<T> = Either<Any?, (T) -> Boolean>
 
-class ResourceSelector<T>(private val project: Project, private val resourceType: Resource<out Collection<T>>) : ComboBox<T>() {
+@Suppress("UNCHECKED_CAST")
+class ResourceSelector<T> @JvmOverloads constructor(
+    private val project: Project,
+    private val resourceType: () -> Resource<out Collection<T>>?,
+    private val autoSelectSingleValue: Boolean = true
+) : ComboBox<T>(MutableCollectionComboBoxModel<T>()) {
+
+    @JvmOverloads
+    constructor(
```

Comment:
how did this new constructor creep in here?

Reply:
It's used for the supplier of the cache provider entry.

Reply:
makes sense